src/manifold/src/manifold.cpp

```diff
@@ -881,4 +882,120 @@ Manifold Manifold::Hull() const { return Hull(GetMesh().vertPos); }
+      // Create AABB around the second triangle, expanded by 2 *
+      // searchLength
+      glm::vec3 minBoundsSecond =
+          glm::min(glm::min(secondMesh.vertPos[secondTriVert.x],
+                            secondMesh.vertPos[secondTriVert.y]),
+                   secondMesh.vertPos[secondTriVert.z]) -
+          glm::vec3(searchLength);
+
+      glm::vec3 maxBoundsSecond =
+          glm::max(glm::max(secondMesh.vertPos[secondTriVert.x],
+                            secondMesh.vertPos[secondTriVert.y]),
+                   secondMesh.vertPos[secondTriVert.z]) +
+          glm::vec3(searchLength);
+
+      // Check for AABB collision
+      if (minBoundsFirst.x <= maxBoundsSecond.x &&
+          maxBoundsFirst.x >= minBoundsSecond.x &&
+          minBoundsFirst.y <= maxBoundsSecond.y &&
+          maxBoundsFirst.y >= minBoundsSecond.y &&
+          minBoundsFirst.z <= maxBoundsSecond.z &&
+          maxBoundsFirst.z >= minBoundsSecond.z) {
+        // AABBs overlap, calculate triangle-to-triangle distance
+        glm::vec3 cp;
+        glm::vec3 cq;
+
+        glm::vec3 p[3] = {firstMesh.vertPos[firstTriVert.x],
+                          firstMesh.vertPos[firstTriVert.y],
+                          firstMesh.vertPos[firstTriVert.z]};
+        glm::vec3 q[3] = {secondMesh.vertPos[secondTriVert.x],
+                          secondMesh.vertPos[secondTriVert.y],
+                          secondMesh.vertPos[secondTriVert.z]};
+
+        float distanceSquared = DistanceTriangleTriangleSquared(cp, cq, p, q);
+
+        minDistanceSquared = std::min(minDistanceSquared, distanceSquared);
+      }
+    }
+  }
+
+  return sqrt(minDistanceSquared);
+}
+
+float Manifold::MinGapCollider(const Manifold& other,
+                               float searchLength) const {
+  // Get intersection
+  auto intersect = *this ^ other;
+  auto prop = intersect.GetProperties();
+
+  // If intersection is non zero, return zero (overlapping)
+  if (prop.volume != 0) {
+    return 0.0f;
+  }
+
+  Vec<Box> faceBox;
+  Vec<uint32_t> faceMorton;
+
+  this->GetCsgLeafNode().GetImpl()->GetFaceBoxMorton(faceBox, faceMorton);
+
+  std::transform(faceBox.begin(), faceBox.end(), faceBox.begin(),
+                 [searchLength](Box box) {
+                   return Box(box.min - glm::vec3(searchLength),
+                              box.max + glm::vec3(searchLength));
+                 });
+
+  Vec<Box> faceBoxOther;
+  Vec<uint32_t> faceMortonOther;
+
+  other.GetCsgLeafNode().GetImpl()->GetFaceBoxMorton(faceBox, faceMorton);
+
+  std::transform(faceBoxOther.begin(), faceBoxOther.end(), faceBoxOther.begin(),
+                 [searchLength](Box box) {
+                   return Box(box.min - glm::vec3(searchLength),
+                              box.max + glm::vec3(searchLength));
+                 });
+
+  Collider collider(faceBox, faceMorton);
+
+  SparseIndices collisions = collider.Collisions(faceBoxOther.cview());
```

Comment:
@elalish @pca006132 
Got a basic implementation working, the test are passing! Now I only need to figure out how to use the collider for the broad phase. Can I get some pointers here?

I looked how the collider is used in [Manifold::Impl::EdgeCollisions](https://github.com/elalish/manifold/blob/master/src/manifold/src/impl.cpp#L868), am I supposed to implement something similar, but for the expanded faces? Also looked at the how the broad phase is implemented [here](https://github.com/elalish/manifold/blob/master/src/manifold/src/boolean3.cpp#L499).

Am I completely off base with this code snippet? Also, I could also use a few pointers on working with SparseIndices, like getting the vertices out of them to calculate the tri-tri distances.

Reply:
looks fine, but I think you forgot to sort them according to the morton code.

Reply:
oh got it, thanks